src/lib/swiss.js

```diff
@@ -100,148 +106,57 @@ function getStandings(options, round, participants, matches) {
-    orderedParticipants.push({ id: null, seed: 0 })
-  }
-  var penalties = generatePenalties(options, orderedParticipants, matches)
-
-  // Split orderParticipants into reasonable chunks
-  var groups = []
-  var divisor = Math.ceil(orderedParticipants.length / 10)
-  var groupSize = orderedParticipants.length / divisor
-  groupSize = groupSize % 2 === 0 ? groupSize : groupSize + 1
-  for (var i = 0; i < orderedParticipants.length; i+= groupSize) {
-    groups.push(orderedParticipants.slice(i, i + groupSize))
+  var mappings = getMappings(participants, matches)
+  if(mappings.length % 2 === 1) {
+    // we simulate the bye having played against every team with a bye
+    // that way those teams will not get a bye again unless the matches are
+    // ridiculously better if they have another
+    // we also want it to bias toward giving byes to teams at the bottom
+    // of the standings
+    mappings.push({id:1337,
+      points:0,
+      opponents:mappings.filter(m => {
+        return m.opponents.length < round - 1
+      }).map( m => {return m.id})
+    })
   }
-
-  var matchups = groups.reduce((matchups, group) => {
-    var best = null
-    for (var list of getAllPairs(group.map(p => p.id))) {
-      var option = list.map(([home, away]) => {
-        var penalty = penalties[home][away] + penalties[away][home]
-        return {
-          penalty: penalty,
-          home: home,
-          away: away
-        }
-      }).reduce((option, pair) => {
-        if (option.penalty === undefined) {
-          option.penalty = 0
-        }
-        if (option.matchups === undefined) {
-          option.matchups = []
-        }
-        option.penalty += pair.penalty
-        option.matchups.push(pair)
-        return option
-      }, {})
-      if (!best || option.penalty < best.penalty) {
-        best = option
-      }
-    }
-
-    return matchups.concat(best.matchups)
-  }, [])
-  matchups.sort((a, b) => {
-    return standings.findIndex(el => el.id === a.home) -
-      standings.findIndex(el => el.id === b.home)
-  })
-  return matchups
-}
-
-function generatePenalties(options, orderedParticipants, matches) {
-  var penalties = orderedParticipants.map((participant, index) => {
-    var opponents = orderedParticipants.slice()
-    opponents.splice(index, 1)
-    var penalties = opponents.map((opponent, idx) => {
-      var penalty = matches.filter(match => {
-        return (
-          match.home.id === participant.id &&
-          match.away.id === opponent.id
-        ) || (
-          match.home.id === opponent.id &&
-          match.away.id === participant.id
+  var arr = []
+  mappings.map(team => {
+    mappings.map(opp => {
+      if(team.id !== opp.id) {
+        arr.push([
+          team.id,
+          opp.id,
+          -1 * (Math.pow(team.points - opp.points, options.standingPower) +
+          options.rematchWeight*team.opponents.reduce((n, o) => {
+            return n + (o === opp.id)
+          }, 0))]
```

Comment:
Possible alternative?

```javascript
var arr = mappings.reduce((a, team, i, orig) => {
  var opps = orig.slice(0, i).concat(orig.slice(i+1))
  for (var opp of opps) {
    b.push([
      team.id,
      opp.id,
      // Not sure about this calculation, seems a little heavy on standings, no?
      // Afraid that this could favor rematches over disparate standing matchups.
      // Also, we probably need to factor in seed to retain uniqueness of comparisons.
      -1 * (Math.pow(team.points - opp.points, options.standingPower) +
        (options.rematchWeight *
          team.opponents.reduce((n, o) => n + (o === opp.id), 0)))
    ])
  }
}, [])
```